### PR TITLE
EVG-2091 improve performance of hosts page

### PR DESCRIPTION
**Pull request by john-m-liu**
The 2 changes are:
- Instead of running 1 db query per host that's running a task to get the task data, get all of it in 1 aggregation
- Stop filtering hosts as you type and instead filter once you hit enter

### CODE REVIEW

File: public/static/js/hosts.js [hunk 1]
```diff
@@ -152,6 +149,12 @@ mciModule.controller('HostsCtrl', function($scope, $filter, $window, $location)
     }
   };
 
+  $scope.onFilterKeyUp = function(event) {
```
Comment (bsamek) [inline public/static/js/hosts.js]: Should this be on key down now?

File: model/host/host.go [hunk 2]
```diff
@@ -35,6 +36,8 @@ type Host struct {
 
 	// the task that is currently running on the host
 	RunningTask string `bson:"running_task,omitempty" json:"running_task,omitempty"`
+	// the full task struct that is running on the host (only populated by certain aggregations)
+	RunningTaskFull []task.Task `bson:"task_full,omitempty" json:"task_full,omitempty"`
```
Comment (bsamek) [inline model/host/host.go]: I don't think this should be an array.

File: model/host/db.go [hunk 3]
```diff
@@ -464,3 +465,21 @@ func statsByDistroPipeline() []bson.M {
...
+			"$match": match,
+		},
+		{
+			"$lookup": bson.M{
+				"from":         task.Collection,
+				"localField":   RunningTaskKey,
+				"foreignField": task.IdKey,
+				"as":           "task_full",
+			},
+		},
```
Comment (bsamek) [inline model/host/db.go]: Should there also be a project stage to reduce the amount of data returned? What fields will we actually use?

File: model/host/host.go (hunk 2 above)
Comment (john-m-liu) [reply model/host/host.go]: fixed

File: public/static/js/hosts.js (hunk 1 above)
Comment (john-m-liu) [reply public/static/js/hosts.js]: changed to keydown

File: model/host/db.go (hunk 3 above)
Comment (john-m-liu) [reply model/host/db.go]: We use 6 fields from the task document in the UI. If we want to remove unneeded fields that we send to the UI we can do it there, but it won't make a noticeable improvement

**Pull request merged**